agent/engine/docker_task_engine.go

```diff
@@ -868,6 +877,13 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(hcerr)}
 	}
 
+	// If LogRouter values are present, check to see if they are the ones expected and
```

Comment:
This should probably be something like: If the container is using a special log driver type "awslogrouter", it means the container wants to use log router to send logs. In this case, override the log driver type to be fluentd and specify appropriate tag and fluentd-address, so that the logs are sent to and routed by the log router.